jupyter_core/paths.py

```diff
@@ -33,6 +34,9 @@ def jupyter_config_dir():
     env = os.environ
     home_dir = get_home_dir()
 
+    if env.get('JUPYTER_NO_CONFIG'):
+        return tempfile.mkdtemp(prefix='jupyter-clean-cfg-')
```

Comment:
If this isn't cached, every call to `jupyter_config_dir` will create a new directory and not clean it up. Maybe store it somewhere?